src/configuration.js

```diff
@@ -41,30 +55,35 @@ export class Configuration {
 
     var rules = defaultRules;
 
-    if (this.options.only.length > 0) {
+    if (this.options.only && this.options.only.length > 0) {
       rules = filterRules(this.options.only);
-    } else if (this.options.except.length > 0) {
+    } else if (this.options.except && this.options.except.length > 0) {
       rules = defaultRules.filter(rule => {
         return (
           this.options.except.map(toUpperCamelCase).indexOf(rule.name) == -1
         );
       });
-    } else {
-      const directory = this.options.searchDirectory || process.cwd();
-      const cosmic = cosmiconfig('graphql-schema-linter', {
-        cache: false,
-        sync: true,
-      }).load(directory);
-
-      if (cosmic) {
-        rules = filterRules(cosmic.config.rules);
-      }
     }
 
     return rules;
   }
 }
 
+function loadOptionsFromConfig(directory) {
+  const cosmic = cosmiconfig('graphql-schema-linter', {
+    cache: false,
+    sync: true,
+  }).load(directory);
+  let options = {};
+  if (cosmic) {
+    // Map config.rules -> `only` param
```

Comment:
Most linters use a `rules` key, so I've adopted that and mapped it to `only`.  Perhaps in a future version we could bump the major version number, drop `except` and change the command line flag to `rules`?


- http://eslint.org/docs/user-guide/configuring#extending-configuration-files
- https://palantir.github.io/tslint/usage/configuration/